android/mobile/app/src/main/java/com/ttd/sso/sample/ui/AuthScreenViewModel.kt

```diff
@@ -0,0 +1,127 @@
+package com.ttd.sso.sample.ui
+
+import android.content.Context
+import android.content.Intent
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import com.myopenpass.auth.OpenPassManager
+import com.myopenpass.auth.OpenPassManagerState
+import com.myopenpass.auth.flow.WebSignInFlowClient
+import com.myopenpass.auth.flow.WebSignInFlowState
+import com.ttd.sso.sample.ui.AuthScreenAction.CheckSignIn
+import com.ttd.sso.sample.ui.AuthScreenAction.SignIn
+import com.ttd.sso.sample.ui.AuthScreenAction.SignOut
+import com.ttd.sso.sample.ui.AuthScreenState.ErrorState
+import com.ttd.sso.sample.ui.AuthScreenState.SignedInState
+import com.ttd.sso.sample.ui.AuthScreenState.SignedOutState
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+sealed interface AuthScreenAction : ViewModelAction {
+    data class SignIn(val context: Context) : AuthScreenAction
+    data class CheckSignIn(val intent: Intent) : AuthScreenAction
+    data object SignOut : AuthScreenAction
+}
```

Comment:
I don't think this is generally useful, I'd get rid of these "actions" and just use plain funs in the VM. Those are easier to use as references which are stable, if invoked from the Compose code.